src/test/spec/unified_runner/test_runner.rs

```diff
@@ -60,58 +64,74 @@ const SKIPPED_OPERATIONS: &[&str] = &[
     "watch",
 ];
 
-pub type EntityMap = HashMap<String, Entity>;
+pub(crate) type EntityMap = HashMap<String, Entity>;
 
-pub struct TestRunner {
-    pub internal_client: TestClient,
-    pub entities: EntityMap,
-    pub fail_point_guards: Vec<FailPointGuard>,
+#[derive(Clone)]
+pub(crate) struct TestRunner {
+    pub(crate) internal_client: TestClient,
+    pub(crate) entities: Arc<RwLock<EntityMap>>,
```

Comment:
the entity map needed to be arc'd + locked so that it could be accessed from all the different test runner tasks. ditto for the failpoint guards.